### PR TITLE
Dynamic donation suggestion

### PR DESCRIPTION
**Pull request by matthewdylan**
#### What's this PR do?
Adds functionality to automatically divide the yearly suggested donation amount by 10 when switching to the monthly option and to multiply the monthly by 10 when switching to the yearly option.

#### Why are we doing this? How does it help us?
We're hoping to simplify the donor process and thus bring in more (mostly monthly) donors.

#### How should this be manually tested?
* Spin up locally
* Visit /donate and make sure the amount doesn't change when switching between monthly and yearly
* Visit "/donate?installmentPeriod=yearly&amount=200&abTesting=True"
* Ensure the monthly amount is 20 and the yearly and one-time amounts are 200
* Visit "/donate?installmentPeriod=monthly&amount=15&abTesting=True"
* Ensure the monthly amount is 15 and the yearly and one-time amounts are 150
* Ensure that changing the amount breaks this automation
    * i.e. switching between monthly/yearly doesn't effect the amount

#### How should this change be communicated to end users?
We'll let membership know.

#### Are there any smells or added technical debt to note?
@ashley-hebler, you had suggested providing an amount for the alternative (vs. automatically dividing/multiplying by 10) but I went with this more automatic route for now since that was the initial intention. I'll talk to membership about the other option (and we could even do both where we default to this automatic way UNLESS there is an alternative value given).

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recAbvCibzXZjDT6G?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Added automated tests? *( )*
* [ ] Tested manually on mobile? *( )*
* [ ] Checked BrowserStack? *( )*
* [ ] Checked for performance implications? *( )*
* [ ] Checked accessibility? *( )*
* [ ] Checked for security implications? *( )*
* [ ] Updated the documentation/wiki? *( )*

#### TODOs / next steps:

* [ ] *your TODO here*

### CODE REVIEW

**Review by ashley-hebler — COMMENTED** [review]
We discussed already, but for some reason I'm running up against some errors on this.

Leaving of misc. notes here too while I have them

File: server/static/js/src/entry/donate/TopForm.vue [hunk 1]
```diff
@@ -227,7 +228,55 @@ export default {
...
+    const getter = this.$store.getters[`${storeModule}/valueByKey`];
+    this.initAmount = getter('amount');
+    this.initFrequency = getter('installment_period');
+    // setting an initial alt amount that is either a tenth of the suggested amount if the init
+    // frequency is yearly/once or ten times the suggested amount if the frequency is monthly
+    // using Math.ceil() here to round up to the nearest whole int
+    if (this.initFrequency === 'monthly') {
+      this.initAltAmount = String(Math.ceil(this.initAmount*10));
+      this.initAltFrequency = ["None", "yearly"];
+    } else {
```
[the remark this answers is not in the document]
Comment (ashley-hebler) [reply server/static/js/src/entry/donate/TopForm.vue:250]: Now that I can test, it does impact `this.initFrequency === 'once'` but maybe we want that

File: server/static/js/src/entry/donate/router.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ function createInitialFormState(queryParams) {
     zipcode = '',
     code = '',
     reason = '',
+    abTesting = false,
```
[the remark this answers is not in the document]
Comment (ashley-hebler) [reply server/static/js/src/entry/donate/router.js:36]: note: probably not a big deal but the true/false value doesn't seem to matter i.e.
http://local.texastribune.org/donate?installmentPeriod=monthly&amount=15&abTesting=False
functions the same as 
http://local.texastribune.org/donate?installmentPeriod=monthly&amount=15&abTesting=True

for me. Same with lowercase true/false

**Review by ashley-hebler — APPROVED** [review]
This is working as expected for me. I left some small notes, but overall this makes sense 🙌

File: server/static/js/src/entry/donate/TopForm.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthewdylan) [reply server/static/js/src/entry/donate/TopForm.vue:250]: Yeah, I think the idea is that in most cases we want the monthly suggested amount to be a fraction of the yearly and the one-time amounts.

File: app.json [hunk 3]
```diff
@@ -3,6 +3,9 @@
   "scripts": {
   },
   "env": {
+    "AB_TESTING_CAMPAIGN_ID": {
+      "required": false
+    },
```
[the remark this answers is not in the document]
Comment (matthewdylan) [reply app.json:8]: Removed this.

File: server/static/js/src/entry/donate/router.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewdylan) [reply server/static/js/src/entry/donate/router.js:36]: I can definitely change the naming.  I'll update to look for true.

**Pull request merged**